datafusion/src/dataframe.rs

```diff
@@ -222,6 +223,21 @@ pub trait DataFrame: Send + Sync {
     /// ```
     async fn collect(&self) -> Result<Vec<RecordBatch>>;
 
+    /// Executes this DataFrame and returns a stream over a single partition
+    ///
+    /// ```
+    /// # use datafusion::prelude::*;
+    /// # use datafusion::error::Result;
+    /// # #[tokio::main]
+    /// # async fn main() -> Result<()> {
+    /// let mut ctx = ExecutionContext::new();
+    /// let df = ctx.read_csv("tests/example.csv", CsvReadOptions::new())?;
+    /// let stream = df.collect_stream().await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    async fn collect_stream(&self) -> Result<SendableRecordBatchStream>;
```

Comment:
What if we called this something like `execute` rather than `collect_stream`? 

```
    async fn execute_stream(&self) -> Result<SendableRecordBatchStream>;
```

This would mirror the naming of `ExecutionPlan::execute` and might make it clearer that `collect` means collect into a Vec and `execute` means get a stream